spring-ai-core/src/main/java/org/springframework/ai/chat/model/AbstractToolCallSupport.java

```diff
@@ -95,6 +97,20 @@ protected Set<String> handleFunctionCallbackConfigurations(FunctionCallingOption
 		return functionToCall;
 	}
 
+	protected List<Message> handleToolCalls(Prompt prompt, ChatResponse response) {
```

Comment:
It's really nice we could abstract this. It might help in the future if we allow "deconstructing" the function calling workflow and have users calling the functions explicitly.

Reply:
Indeed I was thinking the same. Maybe the AbstractToolCallSuppor is not the perfect abstraction for direct usage but we will figure it out.